crates/lumina_shared/src/player/weapon.rs

```diff
@@ -289,5 +338,22 @@ impl WeaponStat {
 }
 
 /// Reload timer based on [`Weapon::reload_duration()`].
-#[derive(Component, Serialize, Deserialize, Deref, DerefMut, Debug, Clone, PartialEq)]
-pub struct WeaponReload(Timer);
+#[derive(Component, Serialize, Deserialize, Debug, Clone, PartialEq)]
+pub struct WeaponReload {
+    pub timer: Timer,           // Reload timer
+    pub bullets_to_reload: u32, // Number of bullets to reload
+}
+
+impl Deref for WeaponReload {
+    type Target = Timer;
+
+    fn deref(&self) -> &Self::Target {
+        &self.timer
+    }
+}
+
+impl DerefMut for WeaponReload {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        &mut self.timer
+    }
+}
```

Comment:
```suggestion
#[derive(Component, Serialize, Deserialize, Deref, DerefMut, Debug, Clone, PartialEq)]
pub struct WeaponReload {
    #[deref]
    pub timer: Timer,           // Reload timer
    pub bullets_to_reload: u32, // Number of bullets to reload
}
```
Bevy has an easier macro system for deref on situations like this